includes/class-plugin.php

```diff
@@ -921,19 +921,16 @@ public function do_admin_actions() {
                 $url = $this->action_link( $action );
 
                 if ( $action === 'flush-cache' ) {
-                    wp_cache_flush()
-                        ? add_settings_error(
-                            'redis-cache',
-                            'flush',
-                            __( 'Object cache flushed.', 'redis-cache' ),
-                            'updated'
-                        )
-                        : add_settings_error(
-                            'redis-cache',
-                            'flush',
-                            __( 'Object cache could not be flushed.', 'redis-cache' ),
-                            'error'
-                        );
+
+                    add_action( 'shutdown', [ $this, 'shutdown_flush_cache' ] );
```

Comment:
We can't rely on using `shutdown` and thus `register_shutdown_function()`, because it 1) may not execute and 2) can swallow errors silently.

Reply:
What eventuality would prevent the `shutdown` hook from firing when calling `/wp-admin/options-general.php?page=redis-cache&action=flush-cache`?

Because you have a catch 22 situation here whereby you need to flush the cache as late in the page load process as possible to prevent subsequent cache requests failing but doing so will miss the `admin-header.php` and `options-head.php` include which is where you are seeing your errors that you set with `add_settings_error()`.

The only solution I can think of is to use a normal `error_log()` command in the `shutdown` hook which you use already in line 992.

Reply:
Yes, I agree, but I thought that you considered Redis Object Cache to be an advanced plugin. That being said I notice that you are saving the settings errors to a transient ...

`
set_transient( 'settings_errors', $messages, 30 );
`

I haven't checked to see what you are doing with that transient but there could be a panel on the settings page to show recent errors and the fact remains that flushing the cache too early causes the issue so if users need to be notified about errors then another way has to be found to notify them other than using the Settings API surely?